### PR TITLE
NO-ISSUE: add vlan and bond nmstate examples

### PR DESCRIPTION
**Pull request by nshidlin**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @nmagnezi 
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/user-guide/restful-api-guide.md [hunk 1]
```diff
@@ -87,4 +87,67 @@ The request will be stored in a temporary file `$request_body` and will be used
 
 ```
 curl -H "Content-Type: application/json" -X POST -d @$request_body ${ASSISTED_SERVICE_URL}/api/assisted-install/v1/clusters/$CLUSTER_ID/downloads/image
+```
+
+## Additional nmstate configuration exmaples
```
Comment (rollandf) [inline docs/user-guide/restful-api-guide.md:92]: typo `examples`

Comment (nshidlin) [reply docs/user-guide/restful-api-guide.md]: done, thanks

File: docs/user-guide/restful-api-guide.md [hunk 2]
```diff
@@ -87,4 +87,67 @@ The request will be stored in a temporary file `$request_body` and will be used
...
+        base-iface: eth0
+        id: 404
+    routes:
+      config:
+      - destination: 0.0.0.0/0
+        next-hop-address: 192.168.143.1
+        next-hop-interface: eth0.404
+        table-id: 254
+```
+### Bonded interfaces
```
[the remark this answers is not in the document]
Comment (nshidlin) [reply docs/user-guide/restful-api-guide.md]: changed to ```Network Bond```

**Pull request merged**